File: Editorial/141_Linked_List_Cycle/Go/1_Hash_Table.go
func hasCycle(head *ListNode) bool {
    nodesSeen := make(map[*ListNode]bool)
    current := head
    for current != nil {
        if nodesSeen[current] {
            return true
        }
        nodesSeen[current] = true
        current = current.Next
    }
    return false
}

File: Editorial/141_Linked_List_Cycle/CPP/1_Hash_Table.cpp
class Solution {
public:
    bool hasCycle(ListNode *head) {
        std::unordered_set<ListNode *> nodesSeen;
        ListNode *current = head;
        while (current != nullptr) {
            if (nodesSeen.find(current) != nodesSeen.end()) {
                return true;
            }
            nodesSeen.insert(current);
            current = current->next;
        }
        return false;
    }
};
